types/datum.go

```diff
@@ -964,17 +964,13 @@ func (d *Datum) convertToString(sc *stmtctx.StatementContext, target *FieldType)
 	case KindBinaryLiteral:
 		s = d.GetBinaryLiteral().ToString()
 	case KindMysqlBit:
-		// issue #25037
-		// bit to binary/varbinary. should consider transferring to uint first.
-		if target.Tp == mysql.TypeString || (target.Tp == mysql.TypeVarchar && target.Collate == charset.CollationBin) {
-			val, err := d.GetBinaryLiteral().ToInt(sc)
-			if err != nil {
-				s = d.GetBinaryLiteral().ToString()
-			} else {
-				s = strconv.FormatUint(val, 10)
-			}
-		} else {
+		// https://github.com/pingcap/tidb/issues/25037.
+		// Consider converting to uint first.
```

Comment:
better add this to here 'https://github.com/pingcap/tidb/pull/31031#issuecomment-1001404832.'